drivers/flash/Kconfig.nrf_mram

```diff
@@ -7,7 +7,7 @@
 config SOC_FLASH_NRF_MRAM
 	bool "Nordic Semiconductor flash driver for MRAM"
 	default y
-	depends on DT_HAS_NORDIC_MRAM_ENABLED
+	depends on DT_HAS_NORDIC_MRAM_ENABLED && !DT_HAS_NORDIC_MRAM_CONTROLLER_ENABLED
```

Comment:
Do we have an sic with both of these nodes enabled? This PR is not adding the new binding to any soc, so its hard to see why we need the ! here :)